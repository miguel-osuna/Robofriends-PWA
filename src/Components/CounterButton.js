import React, { Component } from "react";

class CounterButton extends Component {
  constructor() {
    super();
    this.state = {
      count: 0
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    console.log(nextProps, nextState);
    if (this.state.count !== nextState.count) {
      return true;
    } else {
      return false;
    }
  }

  updateCounter = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  };

  render() {
    const { count } = this.state;
    const { updateCounter } = this;
    console.log("CounterButton");
    return (
      <button color={this.props} onClick={updateCounter}>
        {" "}
        Count: {count}
      </button>
    );
  }
}

export default CounterButton;
