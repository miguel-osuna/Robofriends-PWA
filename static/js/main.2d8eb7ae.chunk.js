(this.webpackJsonprobofriends=this.webpackJsonprobofriends||[]).push([[0],{19:function(e,t,n){e.exports=n(32)},30:function(e,t,n){},31:function(e,t,n){},32:function(e,t,n){"use strict";n.r(t);var o=n(0),r=n.n(o),a=n(9),c=n.n(a),i=(n(24),n(11)),s=n(8),l=(n(29),n(10)),u=n(18);function d(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);t&&(o=o.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,o)}return n}function h(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?d(n,!0).forEach((function(t){Object(u.a)(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):d(n).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}var p={searchField:""},f={is_pending:!1,robots:[],error:""},b=Object(s.d)(Object(s.c)({searchRobots:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:p,t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:{};switch(t.type){case"CHANGE_SEARCH_FIELD":e=h({},e,{searchField:t.payload})}return e},requestRobots:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:f,t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:{};switch(t.type){case"REQUEST_ROBOTS_PENDING":e=h({},e,{is_pending:!0});break;case"REQUEST_ROBOTS_SUCCESS":e=h({},e,{is_pending:!1,robots:t.payload});break;case"REQUEST_ROBOTS_FAILED":e=h({},e,{is_pending:!1,error:t.payload})}return e}}),Object(s.a)(l.a,l.a)),m=(n(30),n(3)),g=n(4),v=n(6),E=n(5),O=n(7),y=function(e){function t(){return Object(m.a)(this,t),Object(v.a)(this,Object(E.a)(t).apply(this,arguments))}return Object(O.a)(t,e),Object(g.a)(t,[{key:"shouldComponentUpdate",value:function(e,t){return!1}},{key:"render",value:function(){return console.log("Header"),r.a.createElement("h1",{className:"f1"}," Robofriends")}}]),t}(o.Component),w=function(e){var t=e.id,n=e.name,a=e.email;return r.a.createElement(o.Fragment,null,r.a.createElement("div",{className:"tc bg-light-blue dib br3 pa3 ma2 grow bw2 shadow-5"},r.a.createElement("img",{src:"https://robohash.org/".concat(t,"?200x200"),alt:"robot"}),r.a.createElement("div",null,r.a.createElement("h2",null,n),r.a.createElement("p",null,a))))},j=function(e){var t=e.robots;return console.log("Card List"),r.a.createElement(o.Fragment,null,r.a.createElement("div",null,t.map((function(e){return r.a.createElement(w,{id:e.id,name:e.name,email:e.email,username:e.username})}))))},R=function(e){var t=e.searchChange;return console.log("SearchBox"),r.a.createElement(o.Fragment,null,r.a.createElement("div",{className:"pa2"},r.a.createElement("input",{onChange:t,className:"pa3 ba b--black bg-lightest-blue",type:"search",placeholder:"Search a robot"})))},S=function(e){return r.a.createElement("div",{style:{overflowY:"scroll",border:"2px solid white",height:"70vh"}},e.children)},C=function(e){function t(){var e;return Object(m.a)(this,t),(e=Object(v.a)(this,Object(E.a)(t).call(this))).updateCounter=function(){e.setState((function(e){return{count:e.count+1}}))},e.state={count:0},e}return Object(O.a)(t,e),Object(g.a)(t,[{key:"shouldComponentUpdate",value:function(e,t){return console.log(e,t),this.state.count!==t.count}},{key:"render",value:function(){var e=this.state.count,t=this.updateCounter;return console.log("CounterButton"),r.a.createElement("button",{color:this.props,onClick:t}," ","Count: ",e)}}]),t}(o.Component),_=(n(31),function(e){function t(){return Object(m.a)(this,t),Object(v.a)(this,Object(E.a)(t).apply(this,arguments))}return Object(O.a)(t,e),Object(g.a)(t,[{key:"componentDidMount",value:function(){this.props.onRequestRobots()}},{key:"render",value:function(){var e=this.props,t=e.searchField,n=e.onSearchChange,a=e.robots,c=e.is_pending,i=a.filter((function(e){return e.name.toLowerCase().includes(t.toLowerCase())}));return c?r.a.createElement("div",null,r.a.createElement("h1",{className:"tc"}," Loading, please wait")):r.a.createElement(o.Fragment,null,r.a.createElement("div",{className:"tc"},r.a.createElement(y,null),r.a.createElement(C,{color:"red"}),r.a.createElement(R,{searchChange:n}),r.a.createElement(S,null,r.a.createElement(j,{robots:i}))))}}]),t}(o.Component)),k=Object(i.b)((function(e){return{searchField:e.searchRobots.searchField,robots:e.requestRobots.robots,is_pending:e.requestRobots.is_pending,error:e.requestRobots.error}}),(function(e){return{onSearchChange:function(t){return e({type:"CHANGE_SEARCH_FIELD",payload:t.target.value})},onRequestRobots:function(){return e((function(e){e({type:"REQUEST_ROBOTS_PENDING"}),fetch("https://jsonplaceholder.typicode.com/users").then((function(e){return e.json()})).then((function(t){return e({type:"REQUEST_ROBOTS_SUCCESS",payload:t})})).catch((function(t){return e({type:"REQUEST_ROBOTS_FAILED",payload:t})}))}))}}}))(_),N=Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));function P(e,t){navigator.serviceWorker.register(e).then((function(e){e.onupdatefound=function(){var n=e.installing;null!=n&&(n.onstatechange=function(){"installed"===n.state&&(navigator.serviceWorker.controller?(console.log("New content is available and will be used when all tabs for this page are closed. See https://bit.ly/CRA-PWA."),t&&t.onUpdate&&t.onUpdate(e)):(console.log("Content is cached for offline use."),t&&t.onSuccess&&t.onSuccess(e)))})}})).catch((function(e){console.error("Error during service worker registration:",e)}))}c.a.render(r.a.createElement("div",null,r.a.createElement(i.a,{store:b},r.a.createElement(k,null))),document.getElementById("root")),function(e){if("serviceWorker"in navigator){if(new URL("/miguel-osuna/Robofriends-PWA",window.location.href).origin!==window.location.origin)return;window.addEventListener("load",(function(){var t="".concat("/miguel-osuna/Robofriends-PWA","/service-worker.js");N?(!function(e,t){fetch(e).then((function(n){var o=n.headers.get("content-type");404===n.status||null!=o&&-1===o.indexOf("javascript")?navigator.serviceWorker.ready.then((function(e){e.unregister().then((function(){window.location.reload()}))})):P(e,t)})).catch((function(){console.log("No internet connection found. App is running in offline mode.")}))}(t,e),navigator.serviceWorker.ready.then((function(){console.log("This web app is being served cache-first by a service worker. To learn more, visit https://bit.ly/CRA-PWA")}))):P(t,e)}))}}()}},[[19,1,2]]]);
//# sourceMappingURL=main.2d8eb7ae.chunk.js.map