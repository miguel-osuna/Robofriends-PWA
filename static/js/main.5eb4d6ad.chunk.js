(this.webpackJsonprobofriends=this.webpackJsonprobofriends||[]).push([[0],{19:function(e,t,n){e.exports=n(32)},30:function(e,t,n){},31:function(e,t,n){},32:function(e,t,n){"use strict";n.r(t);var r=n(0),o=n.n(r),a=n(9),c=n.n(a),i=(n(24),n(11)),l=n(8),s=(n(29),n(10)),u=n(18);function h(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function p(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?h(n,!0).forEach((function(t){Object(u.a)(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):h(n).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}var b={searchField:""},d={is_pending:!1,robots:[],error:""},m=Object(l.d)(Object(l.c)({searchRobots:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:b,t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:{};switch(t.type){case"CHANGE_SEARCH_FIELD":e=p({},e,{searchField:t.payload})}return e},requestRobots:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:d,t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:{};switch(t.type){case"REQUEST_ROBOTS_PENDING":e=p({},e,{is_pending:!0});break;case"REQUEST_ROBOTS_SUCCESS":e=p({},e,{is_pending:!1,robots:t.payload});break;case"REQUEST_ROBOTS_FAILED":e=p({},e,{is_pending:!1,error:t.payload})}return e}}),Object(l.a)(s.a,s.a)),f=(n(30),n(3)),E=n(4),O=n(6),g=n(5),v=n(7),y=function(e){function t(){return Object(f.a)(this,t),Object(O.a)(this,Object(g.a)(t).apply(this,arguments))}return Object(v.a)(t,e),Object(E.a)(t,[{key:"shouldComponentUpdate",value:function(e,t){return!1}},{key:"render",value:function(){return console.log("Header"),o.a.createElement("h1",{className:"f1"}," Robofriends")}}]),t}(r.Component),j=function(e){var t=e.id,n=e.name,a=e.email;return o.a.createElement(r.Fragment,null,o.a.createElement("div",{className:"tc bg-light-blue dib br3 pa3 ma2 grow bw2 shadow-5"},o.a.createElement("img",{src:"https://robohash.org/".concat(t,"?200x200"),alt:"robot"}),o.a.createElement("div",null,o.a.createElement("h2",null,n),o.a.createElement("p",null,a))))},S=function(e){var t=e.robots;return console.log("Card List"),o.a.createElement(r.Fragment,null,o.a.createElement("div",null,t.map((function(e){return o.a.createElement(j,{id:e.id,name:e.name,email:e.email,username:e.username})}))))},C=function(e){var t=e.searchChange;return console.log("SearchBox"),o.a.createElement(r.Fragment,null,o.a.createElement("div",{className:"pa2"},o.a.createElement("input",{onChange:t,className:"pa3 ba b--black bg-lightest-blue",type:"search",placeholder:"Search a robot"})))},R=function(e){return o.a.createElement("div",{style:{overflowY:"scroll",border:"2px solid white",height:"70vh"}},e.children)},w=function(e){function t(){var e;return Object(f.a)(this,t),(e=Object(O.a)(this,Object(g.a)(t).call(this))).updateCounter=function(){e.setState((function(e){return{count:e.count+1}}))},e.state={count:0},e}return Object(v.a)(t,e),Object(E.a)(t,[{key:"shouldComponentUpdate",value:function(e,t){return console.log(e,t),this.state.count!==t.count}},{key:"render",value:function(){var e=this.state.count,t=this.updateCounter;return console.log("CounterButton"),o.a.createElement("button",{color:this.props,onClick:t}," ","Count: ",e)}}]),t}(r.Component),_=(n(31),function(e){function t(){return Object(f.a)(this,t),Object(O.a)(this,Object(g.a)(t).apply(this,arguments))}return Object(v.a)(t,e),Object(E.a)(t,[{key:"componentDidMount",value:function(){this.props.onRequestRobots()}},{key:"render",value:function(){var e=this.props,t=e.searchField,n=e.onSearchChange,a=e.robots,c=e.is_pending,i=a.filter((function(e){return e.name.toLowerCase().includes(t.toLowerCase())}));return c?o.a.createElement("div",null,o.a.createElement("h1",{className:"tc"}," Loading, please wait")):o.a.createElement(r.Fragment,null,o.a.createElement("div",{className:"tc"},o.a.createElement(y,null),o.a.createElement(w,{color:"red"}),o.a.createElement(C,{searchChange:n}),o.a.createElement(R,null,o.a.createElement(S,{robots:i}))))}}]),t}(r.Component)),k=Object(i.b)((function(e){return{searchField:e.searchRobots.searchField,robots:e.requestRobots.robots,is_pending:e.requestRobots.is_pending,error:e.requestRobots.error}}),(function(e){return{onSearchChange:function(t){return e({type:"CHANGE_SEARCH_FIELD",payload:t.target.value})},onRequestRobots:function(){return e((function(e){e({type:"REQUEST_ROBOTS_PENDING"}),fetch("https://jsonplaceholder.typicode.com/users").then((function(e){return e.json()})).then((function(t){return e({type:"REQUEST_ROBOTS_SUCCESS",payload:t})})).catch((function(t){return e({type:"REQUEST_ROBOTS_FAILED",payload:t})}))}))}}}))(_);Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));c.a.render(o.a.createElement("div",null,o.a.createElement(i.a,{store:m},o.a.createElement(k,null))),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[19,1,2]]]);
//# sourceMappingURL=main.5eb4d6ad.chunk.js.map