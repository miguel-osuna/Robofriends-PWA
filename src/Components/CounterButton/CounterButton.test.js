import "../../setupTests.js";
import { shallow } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton.js";
import { jsxEmptyExpression } from "@babel/types";

it("Expect to render CounterButton component", () => {
  const mockColor = "red";
  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it("Correctly increments the counter", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  wrapper.setState({ count: 1 });
  wrapper.find("button").simulate("click");
  expect(wrapper.state("count")).toEqual(2);
  expect(wrapper.props().color).toEqual("red");
});
