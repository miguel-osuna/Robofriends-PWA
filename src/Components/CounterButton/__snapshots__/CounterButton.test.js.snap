// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render CounterButton component 1`] = `
<button
  color="red"
  onClick={[Function]}
>
   
  Count: 
  0
</button>
`;
